src/Symfony/Component/DependencyInjection/Attribute/AutowireInline.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+use Symfony\Component\DependencyInjection\Loader\Configurator\AbstractConfigurator;
+use Symfony\Component\DependencyInjection\Loader\Configurator\InlineServiceConfigurator;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Attribute to tell a parameter to inline a service with custom parameters.
+ *
+ * @author Ismail Özgün Turan <oezguen.turan@dadadev.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class AutowireInline extends Autowire implements BuildDefinitionInterface
+{
+    /**
+     * @param class-string|null $class
+     * @param array             $params Static parameters to pass to instance creation
+     * @param bool|class-string $lazy   Whether to use lazy-loading for this argument
+     */
+    public function __construct(
+        private readonly ?string $class = null,
+        private readonly null|string|array $factory = null,
+        private readonly array $params = [],
+        bool|string $lazy = false,
+    ) {
+        if (!(null !== $class xor null !== $factory)) {
+            throw new LogicException('#[AutowireInline] attribute must declare exactly one of $class or $factory.');
+        }
+
+        if (
+            \is_array($factory)
+            && false === \is_string($factory[0])
```

Comment:
```suggestion
            && !\is_string($factory[0])
            && null !== $factory[0]
```

it should be possible to put `null` here, this allows using the class of the definition instead